test/test_mahalanobis_mixin.py

```diff
@@ -167,3 +167,47 @@ def test_embed_is_linear(estimator, build_dataset):
                             model.transform(X[10:20]))
   assert_array_almost_equal(model.transform(5 * X[:10]),
                             5 * model.transform(X[:10]))
+
+
+@pytest.mark.parametrize('estimator, build_dataset', metric_learners,
+                         ids=ids_metric_learners)
+def test_get_metric_equivalent_to_transform_and_euclidean(estimator,
+                                                          build_dataset):
+  """Tests that the get_metric method of mahalanobis metric learners is the
+  euclidean distance in the transformed space
+  """
+  rng = np.random.RandomState(42)
+  input_data, labels, _, X = build_dataset()
+  model = clone(estimator)
+  set_random_state(model)
+  model.fit(input_data, labels)
+  metric = model.get_metric()
+  n_features = X.shape[1]
+  a, b = (rng.randn(n_features), rng.randn(n_features))
+  euc_dist = euclidean(model.transform(a[None]), model.transform(b[None]))
+  assert (euc_dist - metric(a, b)) / euc_dist < 1e-15
```

Comment:
Or you could use `assert_almost_equal` with a custom tolerance. `1e-15` may be a bit brittle?

Reply:
Yes I agree it's better to use a built-in function. I just saw that `assert_almost_equal` gives an absolute error whereas `numpy.testing.assert_allclose` gives a relative error, I guess it's even better to put a relative one in this case ? For the 1e-15, I agree, it's just that I was thinking this way, if we change our implementation and the precision worsens for one reason or another, we could notice it with this test (and it could help solve bugs elsewhere for instance)

Reply:
yes relative error is probably better